renderdoc/driver/vulkan/vk_serialise.cpp

```diff
@@ -3418,11 +3418,11 @@ void DoSerialise(SerialiserType &ser, ImageLayouts &el)
     SERIALISE_MEMBER(queueFamilyIndex);
   }
   SERIALISE_MEMBER(subresourceStates);
-  SERIALISE_MEMBER(layerCount);
-  SERIALISE_MEMBER(levelCount);
-  SERIALISE_MEMBER(sampleCount);
-  SERIALISE_MEMBER(extent);
-  SERIALISE_MEMBER(format);
+  ser.Serialise(STRING_LITERAL("layerCount"), el.imageInfo.layerCount);
+  ser.Serialise(STRING_LITERAL("levelCount"), el.imageInfo.levelCount);
+  ser.Serialise(STRING_LITERAL("sampleCount"), el.imageInfo.sampleCount);
+  ser.Serialise(STRING_LITERAL("extent"), el.imageInfo.extent);
+  ser.Serialise(STRING_LITERAL("format"), el.imageInfo.format);
```

Comment:
I think it's better to move this into a sub-structure. As long as the members don't change then this will be backwards compatible as the binary format is defined as a linear sequence of bytes with no padding/alignment. So a sub-struct is serialised identically to having those elements linear in the parent struct.

Reply:
Yes, I was worried about backward compatibility. I switched this to serialize as a substructure, and verified that this does seem to load old captures just fine.